File: Server/src/main/core/game/consumable/ConsumableEffect.java
package core.game.consumable;


import core.game.node.entity.player.Player;

/**
 * The type Consumable effect.
 */
public abstract class ConsumableEffect {
    /**
     * Activate.
     *
     * @param p the p
     */
    public abstract void activate(Player p);

    /**
     * Gets health effect value.
     *
     * @param player the player
     * @return the health effect value
     */
    public int getHealthEffectValue(Player player) {
		return 0;
	}
}

File: Emulator/src/main/core/game/consumable/ConsumableEffect.kt
package core.game.consumable

import core.game.node.entity.player.Player

/**
 * Consumable effect.
 */
abstract class ConsumableEffect {
    /**
     * Activate.
     *
     * @param player the player
     */
    abstract fun activate(player: Player)

    /**
     * Gets health effect value.
     *
     * @param player the player
     * @return the health effect value
     */
    open fun getHealthEffectValue(player: Player): Int {
        return 0
    }
}
